### DISCUSSION TITLE
No preview versions of Winget

### DISCUSSION THREAD

**Discussion by fboesmans** [Romanitho/Winget-Install#106] [Ideas]
Hello Romanitho

In the 'Install-Prerequisites' function, your script checks if the latest version of Winget is installed on the device. This check also includes the preview versions of Winget.
Is it an option to filter out the preview versions so that only the full releases can be installed using your script?

Many thanks in advanced!

**Comment by Romanitho** [comment]
Hi,
the Install-Prerequisites only picks the latest stable version.

**Reply by KnifMelti** [reply]
...as long as Microsoft doesn't make an error (has happend in regard to this)
It checks against [https://api.github.com/repos/microsoft/winget-cli/releases/latest](https://api.github.com/repos/microsoft/winget-cli/releases/latest) where previews are banned

**Reply by Romanitho** [reply]
that only takes stable versions in theory

**Reply by fboesmans** [reply]
We have a few clients with the preview version installed. I checked the logs of your script and saw this:

![image](https://github.com/Romanitho/Winget-Install/assets/104124972/39661dc4-2797-4e1d-ae75-741a5d173cc6)

**Reply by Romanitho** [reply]
it doesn't tell if the preview version was installed by our script.

**Reply by KnifMelti** [reply]
Well then (Dec 18, 2023 when that preview came out) maybe Microsoft made a mistake and put it in the latest url. (or something else regarding 2/29/2024 in your log)

**Comment by fboesmans** [comment]
I looked at the logs again and I think I was wrong in my conclusion that it was due to this script. 
I think Microsoft made a mistake and possibly through the WAU the app installer on some devices was updated to the preview version.
